libvast/vast/format/json.hpp

```diff
@@ -99,129 +95,4 @@ class reader final : public multi_layout_reader {
   mutable size_t num_lines_ = 0;
 };
 
-// -- implementation ----------------------------------------------------------
```

Comment:
Note for the reviewer: All of this moved into the `.cpp` file because it's no longer templated. I did not make any changes to the logic outside of the reader constructor, which parses the new option and creates the selector now.